src/solvers/flattening/arrays.cpp

```diff
@@ -535,8 +535,8 @@ void arrayst::add_array_constraints_with(
 
     if(index_expr.type()!=value.type())
     {
-      std::cout << expr.pretty() << '\n';
-      assert(false);
+      prop.error() << expr.pretty() << messaget::eom;
```

Comment:
@rjmunro Could this be why the pretty was dumped in the middle of the json (or is this old news?)